src/js/renderChallenges.js

```diff
@@ -1,9 +1,28 @@
-export default async function renderChallenges(loadData, filterFunction=function(data){return data}, createCardFunction, challengeContainer) {
-    let data = await loadData();
-    data = filterFunction(data);
-    data.forEach((challenge) => {
+async function renderChallenges(data, filterFunction=function(data){return data}, createCardFunction, challengeContainer) {
+    let challenges = await data;
+    challenges = await filterFunction(challenges);
+    challenges.forEach((challenge) => {
       const card = createCardFunction(challenge);
       challengeContainer.append(card);
     });
-    return data;
-  }
\ No newline at end of file
+  }
+
+  async function filterAndUpdateCards(data, renderFunction, challengeContainer, ...filterFunctions) {
+    let challenges = await data;
```

Comment:
I don't understand why this function needs to be async. Is there any case where it will be used before the data has been loaded? To me, the `await data` feels weird, because if it's not loaded by the time this function is called, I would not call it `data`.

i would have refactored all the code to separate the asynchronous "preloading" state from the state of the program in which all data has been already loaded, and all code hence can be synchronous.